src/window.c

```diff
@@ -6383,6 +6383,7 @@ win_fix_scroll(int resize)
 		wp->w_cursor.lnum = lnum;
 	    }
 	    invalidate_botline_win(wp);
+	    wp->w_valid &= ~VALID_CROW;
```

Comment:
Simply invalidating VALID_CROW fixes the issue. But since invalidating it is only necessary for this callback case, I wonder if there is a better place to invalidate it without needlessly making comp_botline() more expensive.